external/vcm/vcm/derived_mapping.py

```diff
@@ -105,27 +113,37 @@ def northward_wind(self):
         return _rotate(self, "x_wind", "y_wind")[1]
 
 
-@DerivedMapping.register("dQu_parallel_to_eastward_wind")
+@DerivedMapping.register(
+    "dQu_parallel_to_eastward_wind", required_inputs=["eastward_wind", "dQu"]
+)
 def dQu_parallel_to_eastward_wind_direction(self):
     sign = np.sign(self["eastward_wind"] / self["dQu"])
     return sign * abs(self["dQu"])
 
 
-@DerivedMapping.register("dQv_parallel_to_northward_wind")
+@DerivedMapping.register(
+    "dQv_parallel_to_northward_wind", required_inputs=["northward_wind", "dQv"]
+)
 def dQv_parallel_to_northward_wind_direction(self):
     sign = np.sign(self["northward_wind"] / self["dQv"])
     return sign * abs(self["dQv"])
 
 
-@DerivedMapping.register("horizontal_wind_tendency_parallel_to_horizontal_wind")
+@DerivedMapping.register(
+    "horizontal_wind_tendency_parallel_to_horizontal_wind",
+    required_inputs=["eastward_wind", "dQu", "northward_wind", "dQv"],
+)
 def horizontal_wind_tendency_parallel_to_horizontal_wind(self):
     tendency_projection_onto_wind = (
         self["eastward_wind"] * self["dQu"] + self["northward_wind"] * self["dQv"]
     ) / np.linalg.norm((self["eastward_wind"], self["northward_wind"]))
     return tendency_projection_onto_wind
 
 
-@DerivedMapping.register("net_shortwave_sfc_flux_derived")
+@DerivedMapping.register(
```

Comment:
(very) Optional suggestion: You could consider adding automatic unit tests for these routines that check `required_inputs` is accurate, or even automaticAlly determine them. This would require A mock "self" object which returns some AppropriAtely-shAped zero ArrAys And logs cAlls to __getitem__, then pAssing it to these functions when they get registered.

This is Also fine though, just thought I'd suggest.

(Apologies for my cApitAl A's, my keyboArd is bugged)